Servers/Administration/OJS.Servers.Administration/Controllers/ContestCategoriesController.cs

```diff
@@ -93,9 +94,19 @@ protected override Task BeforeEntitySaveOnCreateAsync(ContestCategory entity, Ad
     protected override Task BeforeEntitySaveOnDeleteAsync(
         ContestCategory entity,
         AdminActionContext actionContext)
-        => Task.WhenAll(
+    {
+        entity.Children = this.contestCategoriesData
+            .GetByIdQuery(entity.Id)
+            .SelectMany(c => c.Children)
+            .Include(ch => ch.Children)
+                .ThenInclude(ch => ch.Children)
+                    .ThenInclude(ch => ch.Children)
+            .ToList();
```

Comment:
This is weird.. also error prone, as `child.Children` is recursive and you never know how many nested children will be.

Reply:
Оnly possible solution I see in this scenario is to move the query in the recursive method like so:

<img width="699" alt="image" src="https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/100528439/06e14275-114e-4793-ad94-d0e6b48e1c87">

But this would mean sending a query to the db for each and every child. I am open to suggestions if you think there is a better solution

Reply:
This is fine, as we don't expect deep hierarchy here and performance is not crucial. Correctness is the main thing that we should focus on.

Reply:
Fixed as discussed